src/cookbook/gradual-compositionapi-migration.md

```diff
@@ -0,0 +1,258 @@
+    } finally {
+      this.isLoading = false;
+    }
+  }
+},
+created() {
+  this.getGitHubProjects()
+}
+```
+
+### After, with Composition API
+
+```js
+import { reactive, toRefs } from '@vue/composition-api'
+
+export function useFetchAPI(api, options) {
+  const state = reactive({
+    response: null,
+    isLoading: false,
+    hasErrors: null
+  })
+
+  const apiCall = async () => {
+    state.isLoading = true
+    try {
+      let res = await fetch(api, options)
+      state.response = await res.json()
+    } catch (error) {
+      state.hasErrors = error
+    } finally {
+      state.isLoading = false
+    }
+  }
+
+  return {
+    ...toRefs(state),
+    apiCall
+  }
+}
+```
+
+And now we can refactor our earlier component like this:
+
+```js
+import { useFetchAPI } from '@/composables/useFetchAPI.js'
+
+export default {
+  data() {
+    return {
+      gitHubData: [],
+      errors: null,
+      loading: false
+    }
+  },
+  methods: {
+    async callGitHub() {
+      const { hasErrors, isLoading, response, apiCall } = useFetchAPI(
+        `https://api.github.com/users/sdras/repos?page=1&per_page=100`,
+        {}
+      )
+      apiCall()
+      this.gitHubData = response
+      this.errors = hasErrors
+      this.loading = isLoading
+    }
+  }
+}
+```
+
+<iframe src="https://codesandbox.io/embed/refactor-fetch-with-composition-api-klyji?fontsize=14&hidenavigation=1&theme=dark"
+     style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+     title="Refactor Fetch with Composition API"
+     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+   ></iframe>
+
+The second example is reusable throughout the application across multiple components, not just for the gitHub API as the first example shows.
+
+[VueUse](https://vueuse.js.org/) is a great resource to explore that covers many of these encapsulated use cases, ready to use in your application.
+
+## Alternative Patterns
+
+Not everything needs to be refactored, though. Use data driven refactoring to guide you and avoid pre-optimizing when it's not necessary. The Options API is still a valid approach that will still allow you to build powerful applications. And in the event you find there's a lot of reusable logics and/or state amongst your application, that might be a good time to consider the Composition API to enhance your application!
+
+Remember: the strength of the Composition API is in:
+
+- Reuse
+- Organization
+- Explicitly stating what's being returned
+- Flexibility
+
+These same flexible traits that make it easy to compose them mean they can be used for many purposes, so it takes some vigilance to provide clarity those who are reading the code what the purpose may be. We suggest refactoring with that type of organization in mind.
```

Comment:
I'm unclear what *'them'* and *'they'* refer to here.

I think there are some missing words here too. I think it needs *'for'* before *'those'* and *'as to'* before *'what'*.

Reply:
Hm, I think this one I will leave as-is, unless I hear from others that it's confusing, I guess I'm unclear on what's unclear :)